wallet/src/Dscp/Wallet/Knit.hs

```diff
@@ -153,16 +155,25 @@ instance AllConstrained (Elem components) '[Wallet, Core] => ComponentCommandPro
         }
     , CommandProc
         { cpName = "send-tx"
-        , cpArgumentPrepare = identity
+        , cpArgumentPrepare = map
+            $ typeDirectedKwAnn "secretkey" tyCryptoKey
+            . typeDirectedKwAnn "out" tyTxOut
         , cpArgumentConsumer = do
-            passPhrase <- getArg tyString "passphrase"
-            secretKey <- getArg tySecretKey "secretkey"
+            passString <- getArgOpt tyString "pass"
+            secretKeyString <- getArg tyCryptoKey "secretkey"
             outs <- getArgSome tyTxOut "out"
-            pure (passPhrase, secretKey, outs)
-        , cpRepr = \(passPhrase, secretKey, outs) -> CommandAction $ \WalletFace{..} -> do
-            pp <- either throwIO return . mkPassPhrase . encodeUtf8 $ passPhrase
-            sk <- either throwIO return . deserialiseOrFail $ BSL.fromStrict secretKey
-            toValue . ValueString . show . txId <$> walletSendTx pp sk outs
+            pure (passString, secretKeyString, outs)
+        , cpRepr = \(passString, secretKeyString, outs) -> CommandAction $ \WalletFace{..} -> do
+            let
+              decodeSK :: Serialise a => ByteString -> IO a
+              decodeSK = either throwIO return . deserialiseOrFail . BSL.fromStrict
```

Comment:
I don't quite understand why `serialise` is used here at all. You should parse the secret key from the string instead of deserialising. It can be done with `fromByteArray`.

Reply:
You're right.
`Encoded SecretKey` is still `serialise`d, though